src/components/CurriculumUnitDetails/CurriculumUnitDetails.tsx

```diff
@@ -0,0 +1,82 @@
+import { FC } from "react";
+
+import Flex from "@/components/Flex";
+import Box from "@/components/Box";
+import { Heading, LI, P, OL } from "@/components/Typography";
+import { TagFunctional } from "@/components/TagFunctional";
+import { Lesson } from "@/components/UnitModal/UnitModal";
+import { Thread } from "@/components/pages/CurriculumInfo/tabs/UnitsTab/UnitsTab";
+import Accordion from "@/components/Accordion";
+
+export type CurriculumUnitDetailsProps = {
+  unitTitle?: string;
+  threads: Thread[];
+  numberOfLessons: number | null;
+  lessons: Lesson[] | null | undefined;
+};
+
+export const CurriculumUnitDetails: FC<CurriculumUnitDetailsProps> = ({
+  threads,
+  numberOfLessons,
+  lessons,
+}) => {
+  const uniqueThreads = new Set<string>();
+  threads.forEach((thread) => {
+    uniqueThreads.add(thread.title);
+  });
```

Comment:
This might be nicer as:
```ts
const threadTitleSet = new Set<string>(threads.map(thread => thread.title));
```

ie you can instantiate the Set from an Array